### PR TITLE
Auto select case

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?183900

From ticket:

> For 2.24, the mobile team is adding an option to "Auto Select Case" in a case list (if there is only one case).  This will take the user directly to the case detail screen (if it exists) or the list of forms (if there is no case detail).  
> There is a bit of HQ work required:
> - Add checkbox to Advanced Modules on the module page (Advanced section) that states "Automatically select case if only one in list".  This option should be disabled (but shown) if app version is less than 2.24
> - For normal modules that set the case type to commcare-user, we should set the "auto select case" option automatically.  This will auto-select the user case (and not show a list).  This does not require displaying any UI or option to the user in the app builder. This should only be enabled if the app version > 2.24

code buddy @nickpell 

I'd appreciate if someone familiar with advanced & user case modules double-checked the logic here ( @kaapstorm / @snopoke ?)

### CODE REVIEW

**Comment by orangejenny** [comment]
Limited the UI to 2.24. Didn't limit the suite changes because having `autoselect="true"` pre-2.24 won't do anything, but it also won't break mobile.

**Comment by kaapstorm** [comment]
Looks good to me. It would be great to have a test to show that when auto_select_case is set on an advanced module, then the datum attribute is set in the form.

**Comment by nickpell** [comment]
conflicts with master

**Comment by orangejenny** [comment]
test failure unrelated, restarted build

**Comment by orangejenny** [comment]
Added a test for this functionality, and fixed up old tests that were failing - @kaapstorm could you take a look at the recent test-fixing commits?

Once this is merged, ICDS would like it deployed to India.

File: corehq/apps/app_manager/tests/data/case_list_form/case-list-form-suite-usercase.xml [hunk 1]
```diff
@@ -137,7 +139,9 @@
     <instance id="commcaresession" src="jr://instance/session"/>
     <session>
       <datum id="case_id_new_suite_test_0" function="uuid()"/>
-      <datum id="usercase_id" function="instance('casedb')/casedb/case[@case_type='commcare-user'][hq_user_id=instance('commcaresession')/session/context/userid][1]/@case_id"/>
+      <datum id="usercase_id"
+      function="instance('casedb')/casedb/case[@case_type='commcare-user'][hq_user_id=instance('commcaresession')/session/context/userid][1]/@case_id"
+      autoselect="true"/>
```
Comment (snopoke) [inline corehq/apps/app_manager/tests/data/case_list_form/case-list-form-suite-usercase.xml]: Why do you need this here? Won't this already be auto-selected?

File: corehq/apps/app_manager/tests/test_case_list_form.py [hunk 2]
```diff
@@ -395,3 +402,4 @@ def test_case_list_form_advanced(self):
         self.module.case_list_form.form_id = self.form.get_unique_id()
 
         self.assertXmlEqual(self.get_xml('case_list_form_advanced_form'), self.form.render_xform())
+    '''
```
Comment (snopoke) [inline corehq/apps/app_manager/tests/test_case_list_form.py]: A bunch of commented out tests.

**Comment by snopoke** [comment]
Don't think this is ready to go yet. There seems to be some inconsistencies in the way `autoselect` is applied (particularly for the usercase).

Also there are still a bunch of tests commented out.

**Comment by orangejenny** [comment]
Gah, I just commented out the tests so that the one I was working on at the time will run faster, meant to undo that. Will fix.

**Comment by czue** [comment]
fyi, you can run any individual test with `./manage.py test appname.TestModuleName.test_method_name` - shouldn't ever have to comment anything out

**Comment by orangejenny** [comment]
Going to do some testing with usercase scenarios.

**Comment by orangejenny** [comment]
Removed usercase-related code so it stops blocking the case-related code, which is what ICDS needs, from being released. https://github.com/dimagi/commcare-hq/commit/1281f43f6d9acd60242188f220a2a4f79e4c9bc6 is the relevant commit.

**Pull request merged**